### ISSUE TITLE
Add spectrogram layer, similar to the waveform graph

### ISSUE THREAD

**Issue by GoogleCodeExporter** [raynebc/editor-on-fire#264]
```
I'm not sure if submitting an issue is the best way to do this, but it allows 
discussion and attachments, so I'm going to go with it.  Also, a short story, 
because I like stories.

I popped in here a long while back, on issue 132, which was the one that 
eventually led to the (awesome) waveform graph.  At that point, I had an idea 
for a different graphical representation (a spectrogram), presented pretty much 
identically to the waveform.  At the time I commented, the waveform graph 
wasn't implemented yet, and my proof-of-concept was very hacky and used 
pre-generated waveforms.

But!  Once I saw the waveform update, I modified the waveform code to instead 
generate a spectrogram, and it actually worked pretty well, but cannibalized 
the waveform graph.

Then this week I checked in here, and was delighted to find that EOF was not 
only still being maintained, but very actively!  I pulled the code I'd written 
a while back off an old hard drive, reworked it to function alongside the 
waveform graph, and it seems to be working just fine.

I've attached a patch based off the latest revision, r1130.  Since it requires 
new libraries to generate the spectrogram, I modified all the platform-specific 
makefiles, but am only set up to test it on Linux.  Judging by appearances, the 
same flags should apply to the other makefiles as well, but I can't be sure I 
didn't break anything.

On that note: since it does introduce another (possibly burdensome) library 
dependency in fftw3, if you think it worth doing I could go in and separate it 
out with some compile flags to make it an optional feature - conditionally stub 
out the functions called in the main code, and have compile flags around the 
fftw3 include.

There's also a quick fix in there to allow EOF to respond to close buttons, 
which it previously didn't.  I had to add another loop around the main loop to 
get it to work right, but it gets the job done just fine.

I've attached diffs/patches (just a patch -p1 < both.diff should do the trick) 
for each feature independently, and then for both combined.

Since I'm a git guy myself, I also mirrored the svn repo on my GitHub.  My 
changes are in the spectrogram branch:
https://github.com/cincodenada/editor-on-fire/commits/spectrogram

...whoops, sorry for the novel.
TL;DR: Look, I added a spectrogram layer and made the close button work!  I 
initially tried a while ago, but revived my code finally got it working right, 
and am quite glad to see that EOF is still being actively developed.  Yay!  
Patches!  Git mirror!  
```

Original issue reported on code.google.com by `cincoden...@gmail.com` on 19 Apr 2013 at 6:43

Attachments:
- [spectrogram.patch](https://storage.googleapis.com/google-code-attachments/editor-on-fire/issue-264/comment-0/spectrogram.patch)
- [close_button.patch](https://storage.googleapis.com/google-code-attachments/editor-on-fire/issue-264/comment-0/close_button.patch)
- [both.patch](https://storage.googleapis.com/google-code-attachments/editor-on-fire/issue-264/comment-0/both.patch)
